circuits/core/events.py

```diff
@@ -136,7 +136,7 @@ def __repr__(self):
 
         data = "%s %s" % (
             ", ".join(repr(arg) for arg in self.args),
-            ", ".join("%s=%s" % (k, repr(v)) for k, v in self.kwargs.items())
+            ", ".join("%s=%r" % (k, v) for k, v in self.kwargs.items())
```

Comment:
```#!python
>>> print("{0:r}".format("foo"))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: Unknown format code 'r' for object of type 'str'
>>> print("%r" % "foo")
'foo'
```

This is why I don't use it I guess; I'm in favour of ``str.format()`` over ``'...' % ...``